cmake/SetupCodeChecks.cmake

```diff
@@ -206,8 +218,104 @@ macro(blt_add_code_checks)
+                                 SRC_FILES         ${_c_sources})
+    endif()
+
 endmacro(blt_add_code_checks)
 
+##-----------------------------------------------------------------------------
+## blt_add_clang_query_target( NAME             <Created Target Name>
+##                          WORKING_DIRECTORY   <Working Directory>
+##                          COMMENT             <Additional Comment for Target Invocation>
+##                          CHECKERS            <If specified, requires a specific set of checkers>
+##                          DIE_ON_MATCH        <If true, matches stop the build>
+##                          SRC_FILES           [FILE1 [FILE2 ...]] )
+##
+## Creates a new target with the given NAME for running clang_query over the given SRC_FILES
+##
+## COMMENT is prepended to the commented outputted by CMake.
+##
+## WORKING_DIRECTORY is the directory that clang_query will be ran.  It defaults to the directory
+## where this macro is called.
+##
+## DIE_ON_MATCH This will make a match cause the build to fail, useful if you're using this in
+##              CI to enforce rules about your code
+##
+## CHECKERS are the static analysis passes to specifically run on the target. Options
+##              (no value)          : run all available static analysis checks found
+##              (checker1:checker2) : run checker1 and checker2
+##              (interpreter)       : run the clang-query interpeter to interactively develop queries
+##
+## SRC_FILES is a list of source files that clang_query will be run on.
+##-----------------------------------------------------------------------------
+macro(blt_add_clang_query_target)
+  if(CLANGQUERY_FOUND)
+
+    ## parse the arguments to the macro
+    set(options)
+    set(singleValueArgs NAME COMMENT WORKING_DIRECTORY DIE_ON_MATCH)
+    set(multiValueArgs SRC_FILES CHECKERS)
+
+    cmake_parse_arguments(arg
+        "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    # Check required parameters
+    if(NOT DEFINED arg_NAME)
+        message(FATAL_ERROR "blt_add_clang_query_target requires a NAME parameter")
+    endif()
+
+    if(NOT DEFINED arg_SRC_FILES)
+        message(FATAL_ERROR "blt_add_clang_query_target requires a SRC_FILES parameter")
+    endif()
+
+    if(DEFINED arg_WORKING_DIRECTORY)
+        set(_wd ${arg_WORKING_DIRECTORY})
+    else()
+        set(_wd ${CMAKE_CURRENT_SOURCE_DIR})
+    endif()
+   
+    set(interactive_target_name interactive_${arg_NAME})
+    # TODO: "FindPython" instead of just using Python
+    set(CLANG_QUERY_HELPER_SCRIPT ${BLT_ROOT_DIR}/cmake/clang-query-wrapper.py)
+    set(CLANG_QUERY_HELPER_COMMAND python ${CLANG_QUERY_HELPER_SCRIPT} --clang-query ${CLANGQUERY_EXECUTABLE} --checker-directories ${BLT_CLANG_QUERY_CHECKER_DIRECTORIES} --compilation-database-path ${CMAKE_BINARY_DIR})
+    if(arg_DIE_ON_MATCH)
+      set(CLANG_QUERY_HELPER_COMMAND ${CLANG_QUERY_HELPER_COMMAND} --die-on-match)
```

Comment:
Can you fix your indenting in this file to conform to 4 spaces.  Seems to not be indented in some and 2 spaces in others.